src/desktop/apps/rss/routes.js

```diff
@@ -80,10 +88,11 @@ export const findSocialEmbeds = article => {
 
 export const maybeFetchSocialEmbed = section => {
   return new Promise((resolve, reject) => {
-    if (section.type !== "social_embed") {
+    const type = section && section.type
+    if (type !== "social_embed") {
       return resolve(section)
     } else {
-      const { url } = section
+      const url = section && section.url
       const service = url.includes("twitter")
         ? "publish.twitter"
         : "api.instagram"
```

Comment:
I just want to verify an assumption I see in this code - 

1. If we get an `undefined` section, or a section not of type `social_embed`, we just skip it (`return resolve(section)`). 👍 
2. For `social_embed` sections, we check the `url`.
3. If it comes from Twitter, we make a request to the Twitter API. 👍 
4. For every other item, we make a request to the Instagram API. 🤔 

Is that last assumption accurate? Should we do a check to make sure it came from instagram before sending it there, and `return resolve(section)` if it doesn't?

Reply:
@pepopowitz 
The answer is Yes. You mentioned a good point. 👍 Maybe, there are some articles are published on Facebook or any unknown social network app. Their type might be considered as `socail_embed`. That might be a problem. Let me update this.

Reply:
The type `social_embed` accepts only two types of links, instagram and twitter, so this looks correct to me.

Reply:
I believe this may be the origin of the failure in the RSS feed, we should address this issue as well to guard against this in the future, we should not be publishing anything without sections: https://artsyproduct.atlassian.net/browse/GROW-873

Reply:
@eessex 
Oh, I see, Thank you for explaining. 🤝